### ISSUE TITLE
[5] Limit size of downloads for users

### ISSUE THREAD

**Issue by orangewolf** [scientist-softserv/louisville-hyku#57]

**Comment by orangewolf** [comment]
# User Story
Download options (size) need to be limited. We want the **full-size** images to be blocked from download directly and indirectly if at all possible (Randy has pointed out how easy it is to work around the presets), and smaller preset download options available from the viewer – no larger than 600 pixels on the long side, or about ½ the size of the smaller default option and 1/10 the size of the larger default option.

# Acceptance
- [ ] Full-size images blocked from download (accounting for both direct and indirect methods)
- [ ] Smaller preset download options available from the viewer no larger than 600 pixels on the long side, or about ½ the size of the smaller default option and 1/10 the size of the larger default option

**Comment by orangewolf** [comment]
Linked Items
---
[User should not have access to high quality downloads.](https://gitlab.com/notch8/louisville-hyku/-/issues/113)

[EXTENSION](https://gitlab.com/notch8/louisville-hyku/-/issues/140)

**Comment by orangewolf** [comment]
Related Merge Requests
---
[125-fix-uv-download-button](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/146)

[125 - address client feedback](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/147)

[125 - downloaded low res label should be jpgs](https://gitlab.com/notch8/louisville-hyku/-/merge_requests/148)

**Comment by orangewolf** [comment]
Alisha July 2022
---
@rachel.howard, right now only admin users have any ability to download full sized images. all other users and visitors can only download the thumbnail. are you saying that you no longer want admin's to be able to download full sized images either?

instead, all visitors and users with any role will have the same capability, which is to download smaller sized images only?

**Comment by orangewolf** [comment]
Rachel July 2022
---
I'm fine with admins being able to download anything (although I'm also fine with it being restricted - they can access even larger, uncompressed files elsewhere). The download option never went away in the UV on our end so that was one issue we're hoping to correct. Also Randy's concerns that it's too easy to work around.

<details><summary>...</summary>

From: gitlab@mg.gitlab.com <gitlab@mg.gitlab.com>
Sent: Monday, July 18, 2022 11:51 AM
To: Howard, Rachel <rachel.howard@louisville.edu>
Subject: Re: Louisville-hyku | Limit size of downloads for users (#125)

CAUTION: This email originated from outside of our organization. Do not click links, open attachments, or respond unless you recognize the sender's email address and know the contents are safe.

Alisha Evans<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Falishaevn&data=05%7C01%7Crachel.howard%40louisville.edu%7C798486188c4945c2abbf08da68d54575%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637937562472426572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=U4zMSZsCwjjR%2FUErWRZI2PggpfUBr8XjQp8%2F3OCFsvM%3D&reserved=0> commented<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1029744445&data=05%7C01%7Crachel.howard%40louisville.edu%7C798486188c4945c2abbf08da68d54575%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637937562472426572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=DDfq02HDvWD1IOQSLNN5N6p%2BCwIOU4s84azPwVoZPFE%3D&reserved=0>:

@rachel.howard<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Frachel.howard&data=05%7C01%7Crachel.howard%40louisville.edu%7C798486188c4945c2abbf08da68d54575%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637937562472426572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=yj7rGiMWKTkc6Ax3vr4mQQ7nK4f%2FRyirfJWtfUj5jpQ%3D&reserved=0>, right now only admin users have any ability to download full sized images. all other users and visitors can only download the thumbnail. are you saying that you no longer want admin's to be able to download full sized images either?

instead, all visitors and users with any role will have the same capability, which is to download smaller sized images only?

-
Reply to this email directly or view it on GitLab<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1029744445&data=05%7C01%7Crachel.howard%40louisville.edu%7C798486188c4945c2abbf08da68d54575%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637937562472426572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=DDfq02HDvWD1IOQSLNN5N6p%2BCwIOU4s84azPwVoZPFE%3D&reserved=0>.
You're receiving this email because you have been mentioned on gitlab.com<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F&data=05%7C01%7Crachel.howard%40louisville.edu%7C798486188c4945c2abbf08da68d54575%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637937562472426572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Q0LbiTyZvU%2BhMfU8qwum6m%2BjOhMrQOq4Xi0GnMAKHS8%3D&reserved=0>. Unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fsent_notifications%2F37df025855fabe5687fd1a9ca1d30d6b%2Funsubscribe&data=05%7C01%7Crachel.howard%40louisville.edu%7C798486188c4945c2abbf08da68d54575%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637937562472426572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=pLHqKpvVcBeDyilSbdHU5V3zH1vLeueVZZXDqPAvlOA%3D&reserved=0> from this thread * Manage all notifications<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fprofile%2Fnotifications&data=05%7C01%7Crachel.howard%40louisville.edu%7C798486188c4945c2abbf08da68d54575%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637937562472426572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=0XsEbmK4Xn4hFk0mF%2BAkMeWJZ9afnFiinLGqD0qtH%2FA%3D&reserved=0> * Help<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fhelp&data=05%7C01%7Crachel.howard%40louisville.edu%7C798486188c4945c2abbf08da68d54575%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637937562472426572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=itkma7%2Bsf4nyRd4aJmoUMt3NIde07Tr3XX2WTS9tfwk%3D&reserved=0>

</details>

**Comment by orangewolf** [comment]
Shana July 2022
---

Is the ask just to make a smaller download size available from the UV download? cc @rachel.howard Also, should users be able to download the entire PDF? Should that different functionality be seen for admin/superadmin vs everyone else?

And/or will we also be updating the options in the actions's download dropdown of the items section, that Alisha previously worked on before?

Dev Notes:



![Image](https://user-images.githubusercontent.com/1054448/203338667-c793dc55-08b4-4fef-9d7c-e659e92c947c.png)



![Image](https://user-images.githubusercontent.com/1054448/203338703-6f901501-8424-44d6-8a02-daa8410a4696.png)



![Image](https://user-images.githubusercontent.com/1054448/203338740-f65c683f-4990-49a8-a4c9-e624c7986933.png)

**Comment by orangewolf** [comment]
Shana July 2022
---
Pointing is based off of the following assumption:
Smaller download available for jpgs in the UV download option
PDF option ???
Permission based access ??

**Comment by orangewolf** [comment]
Rachel July 2022
---
We're actually fine with users being able to download an entire PDF for text-based items. Do want smaller JPEG options from the UV. Items (available to logged-in admin users only) do not have to change from Alisha's previous work, unless it's easier to do that to make the UV changes function, in which case it's fine, because admins have access to full-resolution files on our servers anyway, and don't need to download them from Hyku.

**Comment by orangewolf** [comment]
Shana August 2022
---
After meeting with Rob and Kirk, the easiest implementation would be to restrict the derivatives to the size you are specifying. Then it should automagically just work everywhere at that size and it prevents work-arounds. 

If the users can get the high res images elsewhere, there is no need for us to store it and take up the extra space required to do so as well. No high res will be available to download or view in hyku. 

Does this sound OK to you @rachel.howard ?

cc @kirkkwang

**Comment by orangewolf** [comment]
Rachel August 2022
---
Yes, that works for me!

<details><summary>...</summary>

From: gitlab@mg.gitlab.com <gitlab@mg.gitlab.com>
Sent: Thursday, August 18, 2022 1:13 PM
To: Howard, Rachel <rachel.howard@louisville.edu>
Subject: Re: Louisville-hyku | [5] Limit size of downloads for users (#125)

CAUTION: This email originated from outside of our organization. Do not click links, open attachments, or respond unless you recognize the sender's email address and know the contents are safe.

Shana Moore<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2FShanaLMoore&data=05%7C01%7Crachel.howard%40louisville.edu%7C05ce4a071b5e450e4a4608da813cee75%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637964395970764935%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=EhdcekUv64ndNl8CEMvlaZPSHOvGW6kMUHf80SEzZBI%3D&reserved=0> commented on a discussion<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1069509518&data=05%7C01%7Crachel.howard%40louisville.edu%7C05ce4a071b5e450e4a4608da813cee75%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637964395970921182%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=9JqPsfqxWZ80aYS%2FhfQO9JcIpV73cEEVyhJFzDvvMGM%3D&reserved=0>:

After meeting with Rob, the easiest implementation would be to restrict the derivatives to the size you are specifying. Then it should automagically just work everywhere at that size and it prevents work-arounds.

If the users can get the high res images elsewhere, there is also no need for us to store it and take up the extra space required to do so as well.

Does this sound OK to you @rachel.howard<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Frachel.howard&data=05%7C01%7Crachel.howard%40louisville.edu%7C05ce4a071b5e450e4a4608da813cee75%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637964395970921182%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=DSw0eG8q1fQBShAkqk%2BAdS%2BR5FWzSHsj1yZ6Um1fLKA%3D&reserved=0> cc @kirkkwang<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fkirkkwang&data=05%7C01%7Crachel.howard%40louisville.edu%7C05ce4a071b5e450e4a4608da813cee75%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637964395970921182%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=5Txqn3UMLNAy%2FRg6NvFmaiylGiyxWw1S%2By9qrDviogY%3D&reserved=0>

-
Reply to this email directly or view it on GitLab<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1069509518&data=05%7C01%7Crachel.howard%40louisville.edu%7C05ce4a071b5e450e4a4608da813cee75%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637964395970921182%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=9JqPsfqxWZ80aYS%2FhfQO9JcIpV73cEEVyhJFzDvvMGM%3D&reserved=0>.
You're receiving this email because you have been mentioned on gitlab.com<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F&data=05%7C01%7Crachel.howard%40louisville.edu%7C05ce4a071b5e450e4a4608da813cee75%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637964395970921182%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=CwATsSTH7ctno2vmLJfdMDk3OlSu1O4a9%2BWIfH6r%2Fts%3D&reserved=0>. Unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fsent_notifications%2F5e46d97cf76dd8aaa62de430da349b00%2Funsubscribe&data=05%7C01%7Crachel.howard%40louisville.edu%7C05ce4a071b5e450e4a4608da813cee75%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637964395970921182%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=fZJZVtPJRR2yUcSm01CrxDzKBRqYRl110lPRFTxGO88%3D&reserved=0> from this thread * Manage all notifications<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fprofile%2Fnotifications&data=05%7C01%7Crachel.howard%40louisville.edu%7C05ce4a071b5e450e4a4608da813cee75%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637964395970921182%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Gn0r1bIB4LTpko5RGYllfqezQ6ZGCSfQk1TC34evUYA%3D&reserved=0> * Help<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fhelp&data=05%7C01%7Crachel.howard%40louisville.edu%7C05ce4a071b5e450e4a4608da813cee75%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637964395970921182%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=jS28fytP4VvMdP7fb%2F%2FPFaMZopyep2622ensPNV9BaI%3D&reserved=0>

</details>

**Comment by orangewolf** [comment]
Kirk August 2022
---
Need some assistance from Rob on this one, `On Hold` until then.

**Comment by orangewolf** [comment]
Kirk August 2022
---
Hi @rachel.howard, the approach of just ingesting a 600px long edge image is severely affecting the OCR accuracy since the resolution is so small so I'm thinking it might not be a viable solution unless it's okay to compromise OCR.  I've been kicking around a different solution where I would hide the download button from the UV and also from the file set actions menu and use the thumbnail download.

<img src="/uploads/a09814aaa28ff6a62f3ec00010d973fc/image.png" width=500>
<img src="/uploads/93308afea0d9888324a93a8a3a82b5c8/image.png" width=500>

This would still leave the tiff, jp2, and pdf versions that are larger than desired.  Are these download options needed?  I can hide them and the user would only see the following:

<img src="/uploads/13c3d8950251bcf2b405d78fe5942c2b/image.png" width=500>

Do you think this be enough to obscure the user from easily accessing the full resolution image?

**Comment by orangewolf** [comment]
Rachel August 2022
---
We definitely didn't intend the 600 pixel version to be the one ingested! We just want to limit what's downloaded. No need for anything to be downloaded from the "Actions" shown above, as back-end users (all 3 of us) have access to the full-resolution images on the servers from which they're imported. End-users can download from the Universal Viewer and we want that to be smaller. I assumed there would be a way to generate smaller options for that, using the same program that generates thumbnails. Including @rtkuehn because he had some qualms about merely obscuring the option (which we have done on our production site - https://hyku.library.louisville.edu) - savvy users could get around that by altering the URL, since the images are in a public spot. Would it help to set up a meeting to discuss this further so we're all on the same page?

**Comment by orangewolf** [comment]
Kirk August 2022
---
Thanks for clearing that up Rachel, I'd love to have a chat about it tomorrow!

**Comment by orangewolf** [comment]
Rachel August 2022
---
Sure! I'm free between 1-3:30 pm EDT.

<details><summary>...</summary>

From: gitlab@mg.gitlab.com <gitlab@mg.gitlab.com>
Sent: Wednesday, August 24, 2022 6:46 PM
To: Howard, Rachel <rachel.howard@louisville.edu>
Subject: Re: Louisville-hyku | [5] Limit size of downloads for users (#125)

CAUTION: This email originated from outside of our organization. Do not click links, open attachments, or respond unless you recognize the sender's email address and know the contents are safe.

Kirk Wang<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fkirkkwang&data=05%7C01%7Crachel.howard%40louisville.edu%7C67b09f58ba6e4f086be208da8622707d%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637969779725830422%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=M1ZEmf386Z6fyMp6EXhaqwNSC%2BQZnqqCpJhJNx0kNNY%3D&reserved=0> commented on a discussion<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1077203384&data=05%7C01%7Crachel.howard%40louisville.edu%7C67b09f58ba6e4f086be208da8622707d%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637969779725830422%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=E21799yMdJjP2kjU2FaIZxunYCuatXzNW1kX%2BiGAgv0%3D&reserved=0>:

Thanks for clearing that up Rachel, I'd love to have a chat about it tomorrow!

-
Reply to this email directly or view it on GitLab<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1077203384&data=05%7C01%7Crachel.howard%40louisville.edu%7C67b09f58ba6e4f086be208da8622707d%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637969779725830422%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=E21799yMdJjP2kjU2FaIZxunYCuatXzNW1kX%2BiGAgv0%3D&reserved=0>.
You're receiving this email because of your account on gitlab.com<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F&data=05%7C01%7Crachel.howard%40louisville.edu%7C67b09f58ba6e4f086be208da8622707d%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637969779725830422%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=V9YHVH092izipD7FCZj%2FGzTf%2FrBpbhlca5%2FlxeKIbKQ%3D&reserved=0>. Unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fsent_notifications%2F170ce0e9a140326b34a29fc23e1db7e8%2Funsubscribe&data=05%7C01%7Crachel.howard%40louisville.edu%7C67b09f58ba6e4f086be208da8622707d%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637969779725830422%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=NB30t49N4TwmYw1DapR1rH8aoH7AaogWW1dx5HqWxwI%3D&reserved=0> from this thread * Manage all notifications<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fprofile%2Fnotifications&data=05%7C01%7Crachel.howard%40louisville.edu%7C67b09f58ba6e4f086be208da8622707d%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637969779725830422%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=K5S2J0HWmoCHEsblZ1BHyZqZrY8sy4L296SeQ5yBhkQ%3D&reserved=0> * Help<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fhelp&data=05%7C01%7Crachel.howard%40louisville.edu%7C67b09f58ba6e4f086be208da8622707d%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637969779725986657%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=iVvcU9cN2QyBFUk3EuZUFsMxKyG8fezLOE%2Brlyk5%2FgQ%3D&reserved=0>

</details>

**Comment by orangewolf** [comment]
Shana August 2022
---
Options: 

1. Original proposal to only make 600px available will not work due to unforeseen issues with OCR. High res is required for text matching. 

2. Only make a 600 jpg available to download. The action drop down download doesn't matter because it's only available for admins. We have acknowledged that savvy users may still be able to find a way to download them from the url. To make that bit foolproof, we recommend implementing point 3.



![Image](https://user-images.githubusercontent.com/1054448/203340346-872fd999-87b4-4acf-9161-b8e6e40dec62.png)





![Image](https://user-images.githubusercontent.com/1054448/203340399-360b6400-0c61-4c1d-8f42-1eb3e9c1531f.png)



3. Watermark high res versions.


We are going with option 2

**Comment by orangewolf** [comment]
Kirk August 2022
---
Tested on staging :white_check_mark:

https://lv-hyku-staging.notch8.cloud/concern/images/884fadd0-9f68-47dd-b09a-7316eff12ba3?locale=en

**Comment by orangewolf** [comment]
Rachel August 2022
---
Cool! The certificate apparently expired today, so I got a warning about that when I accessed staging.

**Comment by orangewolf** [comment]
Rachel August 2022
---
Reminder that PDF download is also fine/desirable, esp. for items that were imported as PDFs in the first place.

Get Outlook for iOS<https://aka.ms/o0ukef>

<details><summary>...</summary>

________________________________
From: gitlab@mg.gitlab.com <gitlab@mg.gitlab.com> on behalf of Kirk Wang (@kirkkwang) <gitlab@mg.gitlab.com>
Sent: Friday, August 26, 2022 1:31:42 PM
To: Howard, Rachel <rachel.howard@louisville.edu>
Subject: Re: Louisville-hyku | [5] Limit size of downloads for users (#125)


CAUTION: This email originated from outside of our organization. Do not click links, open attachments, or respond unless you recognize the sender's email address and know the contents are safe.

Kirk Wang<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fkirkkwang&data=05%7C01%7Crachel.howard%40louisville.edu%7C52882cfbf81e450e508208da8788d8ad%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637971319071060294%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=nxZ%2Bu6k2dr%2FPugX8rs%2Bi4lsoP7tmQr4oKYnNURFc8T4%3D&reserved=0> commented<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1079683243&data=05%7C01%7Crachel.howard%40louisville.edu%7C52882cfbf81e450e508208da8788d8ad%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637971319071060294%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=A%2BZbIhX%2BxkC3xFDF1OrP6Ge1Ox%2BiyfGv6RgnqGAgcBE%3D&reserved=0>:

Tested on staging ✅

—
Reply to this email directly or view it on GitLab<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1079683243&data=05%7C01%7Crachel.howard%40louisville.edu%7C52882cfbf81e450e508208da8788d8ad%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637971319071060294%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=A%2BZbIhX%2BxkC3xFDF1OrP6Ge1Ox%2BiyfGv6RgnqGAgcBE%3D&reserved=0>.
You're receiving this email because of your account on gitlab.com<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F&data=05%7C01%7Crachel.howard%40louisville.edu%7C52882cfbf81e450e508208da8788d8ad%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637971319071060294%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=srrGcDEFT%2FR%2BAUjB66%2BzjFb8SEFzoy8CeaBD2SCHSag%3D&reserved=0>. Unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fsent_notifications%2F8d9f3da38b3b8efadf6c9cf432babda6%2Funsubscribe&data=05%7C01%7Crachel.howard%40louisville.edu%7C52882cfbf81e450e508208da8788d8ad%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637971319071060294%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=BOJsGa18ENxkUTmJnwq68RqnJ5JExvwaTjcff733CzY%3D&reserved=0> from this thread · Manage all notifications<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fprofile%2Fnotifications&data=05%7C01%7Crachel.howard%40louisville.edu%7C52882cfbf81e450e508208da8788d8ad%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637971319071060294%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=tlDSTVoFA20NBhZpNwtWm3yp74lOohKHaw9P239FJMk%3D&reserved=0> · Help<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fhelp&data=05%7C01%7Crachel.howard%40louisville.edu%7C52882cfbf81e450e508208da8788d8ad%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637971319071060294%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=%2BwttigQOKgJ5x8rjLSNIMXzN1qxz7scaHEH9JRQA6uo%3D&reserved=0>

</details>

**Comment by orangewolf** [comment]
Kirk August 2022
---
Yes! Thanks for confirming that. I left that option in there. I removed the high res option and also had to remove the option where when you zoom in, you get this "current view" option to make it work. Are you still able to access staging? I wasn't sure what you meant able the certificate thing.

**Comment by orangewolf** [comment]
Rachel August 2022
---
I had to go via Firefox (and click through lots of warnings) but could get there. The https certificate must’ve expired. Chrome blocks access entirely when that happens.

Get Outlook for iOS<https://aka.ms/o0ukef>

<details><summary>...</summary>

________________________________
From: gitlab@mg.gitlab.com <gitlab@mg.gitlab.com> on behalf of Kirk Wang (@kirkkwang) <gitlab@mg.gitlab.com>
Sent: Saturday, August 27, 2022 9:10:03 AM
To: Howard, Rachel <rachel.howard@louisville.edu>
Subject: Re: Louisville-hyku | [5] Limit size of downloads for users (#125)


CAUTION: This email originated from outside of our organization. Do not click links, open attachments, or respond unless you recognize the sender's email address and know the contents are safe.

Kirk Wang<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fkirkkwang&data=05%7C01%7Crachel.howard%40louisville.edu%7C54ab5ef58ea744e7e93508da882d7628%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637972026088176382%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=E6eNKX7WUrpV9q2gi6oy2jLCrc2QkC9tpI1Q8iVwLrY%3D&reserved=0> commented on a discussion<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1080030640&data=05%7C01%7Crachel.howard%40louisville.edu%7C54ab5ef58ea744e7e93508da882d7628%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637972026088176382%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=QWxuhlBYlOBw8KDwQrNlo%2BIVmFztaGAusNGGZ5i3qx8%3D&reserved=0>:

Yes! Thanks for confirming that. I left that option in there. I removed the high res option and also had to remove the option where when you zoom in, you get this "current view" option to make it work. Are you still able to access staging? I wasn't sure what you meant able the certificate thing.

—
Reply to this email directly or view it on GitLab<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fnotch8%2Flouisville-hyku%2F-%2Fissues%2F125%23note_1080030640&data=05%7C01%7Crachel.howard%40louisville.edu%7C54ab5ef58ea744e7e93508da882d7628%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637972026088176382%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=QWxuhlBYlOBw8KDwQrNlo%2BIVmFztaGAusNGGZ5i3qx8%3D&reserved=0>.
You're receiving this email because of your account on gitlab.com<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F&data=05%7C01%7Crachel.howard%40louisville.edu%7C54ab5ef58ea744e7e93508da882d7628%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637972026088176382%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Yqyip1i7qswaewOdy3XWh4yDgYJyqU1bhOugbiAS9Dg%3D&reserved=0>. Unsubscribe<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fsent_notifications%2F068567888f589c6ae8b236713da4f878%2Funsubscribe&data=05%7C01%7Crachel.howard%40louisville.edu%7C54ab5ef58ea744e7e93508da882d7628%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637972026088176382%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=a4GvjjkoVit0Vej89GA%2FwsLyXIXUmr%2BFIkOttt1ZHug%3D&reserved=0> from this thread · Manage all notifications<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2F-%2Fprofile%2Fnotifications&data=05%7C01%7Crachel.howard%40louisville.edu%7C54ab5ef58ea744e7e93508da882d7628%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637972026088176382%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Y5s4ogBMN0ylGLVj5aUIDJINjd4FUIJPWKe26v5ZJMM%3D&reserved=0> · Help<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgitlab.com%2Fhelp&data=05%7C01%7Crachel.howard%40louisville.edu%7C54ab5ef58ea744e7e93508da882d7628%7Cdd246e4a54344e158ae391ad9797b209%7C0%7C0%7C637972026088332629%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=bbgtQczZyWflH6fx3%2B%2FUMS6Hebye%2F%2BE8Aa9Ohc%2B6WdY%3D&reserved=0>

</details>

**Comment by orangewolf** [comment]
Kirk August 2022
---
This is how it looks on my firefox


https://gitlab.com/notch8/louisville-hyku/uploads/f954a3a6cc256fd5a7a4fdc80451106f/lv-cert.gif

**Comment by orangewolf** [comment]
Rachel August 2022
---


![Image](https://user-images.githubusercontent.com/1054448/203342357-9092878b-4775-4922-8bff-025f8a62641c.png)
The code is pushed to our dev server, though, so I'll just test it there.

**Comment by orangewolf** [comment]
Shana August 2022
---
@kirkkwang to chime in here, I'm not sure if this detail has been shared with you yet. 

Yes we can/should resolve the issue Rachel is seeing. Most likely cc @aprilrieger would be our POC for this to update our SSL cert. The bit that I think I've failed to mention is that LV also has their own dev server, and they host their own prod. So when we push our code into client QA we also need to push our updated code to their github and let Randy know. (I did this for you last week). 

Although having clients testing on our dev server is preferred, because we have control of it and can have access to the examples they use (esp if something doesn't pass client QA), for various other reasons this client also likes to test on their dev server.

**Comment by orangewolf** [comment]
Rachel August 2022
---
The code to limit download size of JPGs works on our docker-dev server - hurray! It currently downloads everything as "default.jpg" - is there a simple way to have it pick up the filename, or would that be out of scope? 

As for PDFs, the option to download the PDF shows up in the viewer for PDF-generated works (for example, Law School World War II News Letters, which are also on staging), but it doesn't download anything. When not logged in the browser responds with: "You are unauthorized to access that file." This appears to be a bug.

**Comment by orangewolf** [comment]
Shana August 2022
---
@rachel.howard The unauthorized access to file is likely something @rtkuehn needs to fix on your server. A permissions issue. I will edit this message in a bit confirming the path it's trying to reach.[@rachel.howard](https://github.com/rachel.howard) The unauthorized access to file is likely something [@rtkuehn](https://github.com/rtkuehn) needs to fix on your server. A permissions issue. I will edit this message in a bit confirming the path it's trying to reach.

**Comment by orangewolf** [comment]
Rachel August 2022
---
It also appears on staging though - see https://lv-hyku-staging.notch8.cloud/concern/texts/85d7c8a6-a676-4131-aa6e-1eaf0c3ee108?locale=en and try to download the PDF from there.

**Comment by orangewolf** [comment]
Shana August 2022
---
Oh thanks for pointing that out. We'll take a look at it either way. I was clicking on the pdf directly and downloading it from there: https://lv-hyku-staging.notch8.cloud/concern/parent/85d7c8a6-a676-4131-aa6e-1eaf0c3ee108/file_sets/1aabef17-5d25-4ecd-8dfb-9d46da6f247c <-- does this work for you? @rachel.howardOh thanks for pointing that out. We'll take a look at it either way. I was clicking on the pdf directly and downloading it from there: https://lv-hyku-staging.notch8.cloud/concern/parent/85d7c8a6-a676-4131-aa6e-1eaf0c3ee108/file_sets/1aabef17-5d25-4ecd-8dfb-9d46da6f247c <-- does this work for you? [@rachel.howard](https://github.com/rachel.howard)

**Comment by orangewolf** [comment]
Shana August 2022
---
Confirming that I am able to download the pdf locally. [DEMO](https://share.getcloudapp.com/04uWe48g)

So perhaps our server needs to have some permissions unblocked as well. I'll report back soon and hopefully Randy can replicate the fix on the client's server.

### Dev Notes

Locally, when I click download it hits:  

`GET "/downloads/7698799e-287e-45a6-837a-9888c396692e" ... Processing by Hyrax::DownloadsController#show`

I also see "Redirected to http://hyku.test/uv/uv.html"

Updatea: 

Besides fixing the SSL cert issue staging, I did not make any changes to the permissions. However, I was able to successfully download rachel's [example](https://lv-hyku-staging.notch8.cloud/concern/texts/85d7c8a6-a676-4131-aa6e-1eaf0c3ee108?locale=en) from above. 

The diff? I'm logged in, so this is not a server issue. It's more likely restricting the user (or non user) type in this controller action? :thinking:  [DEMO](https://share.getcloudapp.com/6quG7yLj)

**Comment by orangewolf** [comment]
Shana August 2022
---
It turns out, we implemented a restriction [here](https://gitlab.com/notch8/louisville-hyku/-/blob/main/app/controllers/hyrax/downloads_controller_decorator.rb#L16). It only allows admin users to access the download path. 

This was done as a part https://gitlab.com/notch8/louisville-hyku/-/issues/113

I believe we can proceed with removing this restriction. In #113 the desired behavior at the time was as follows: 

* admin/ super admin users can access the actual file and its derivatives using the downloads path
* non admin users and logged out visitors can see the thumbnail of a file, but can't access the full res file or any of the other derivatives

I would consider this ticket an expansion to and a rework of the original implementation, so I believe we can revert this without undesired consequences.

**Comment by orangewolf** [comment]
Diem August 2022
---
**This passes Notch8 QA:** See attached screen recording.

tested on:
- https://lv-hyku-staging.notch8.cloud/concern/texts/85d7c8a6-a676-4131-aa6e-1eaf0c3ee108?locale=en
- Firefox and Chrome
- logged in and logged out

<details><summary>Logged in and logged out users can download both JPG and PDF files with the correct file name</summary>
![Screen_Recording_2022-08-31_at_11.26.24_AM](/uploads/b422c207c1909315a87abcbd8e24b737/Screen_Recording_2022-08-31_at_11.26.24_AM.mp4)![Screen_Recording_2022-08-31_at_11.27.54_AM](/uploads/b3ecdb95a136c968208fa0c456b8260f/Screen_Recording_2022-08-31_at_11.27.54_AM.mp4)
</details>

**Comment by orangewolf** [comment]
Rachel August 2022
---
Looks perfect!

**Comment by orangewolf** [comment]
Rachel September 2022
---
The code works on our server too! Only remaining question - when/why did the contents panel stop automatically opening in the UV? We really liked that.

**Comment by orangewolf** [comment]
Kirk September 2022
---
@rachel.howard 

It looks like that when `_universel_viewer.html.erb` was brought back in f13b1e38, the Universal Viewer now reads from the `uv-config.json` from this project. The `panelOpen` setting was set to `false` and now has been changed. There are some other settings in there that are opposite to what I've seen as a default option from this sandbox UV version: http://wellcometesters.azurewebsites.net/ such as `"defaultToTreeEnabled": false,` and `"expandFullEnabled": true,`. I flipped them to test but saw no change, at least with the test data I had so I did not alter those values.

I just made the merge request to make the panel open automatically and should be pushed over if all goes well in testing.

<img src="/uploads/a1516e38e0cc3fd8543dd120f3371bbe/lv125-panel.gif" width="500">
